### PR TITLE
fix: Fix voice commands actions in voice prompt window

### LINKED ISSUE #2171: Go to chat Activity as soon as voice command is pressed
**Describe the bug**
Currently, the voice commands even after clicked waits until the speech recognition timesout.
This shouldn't happen. It should go to the ChatActivity otherwise it doesn't justify the purpose of these commands. 

**Expected Behaviour**
The voice commands should go to chatAcitivity and display results as soon as they are clicked.

**Would you like to work on the issue?**
Yes

### PR DESCRIPTION
**Pull request by atm1504**
Fixes #2171 #2241 

Screenshots for the change: 
ezgif com-video-to-gif

### CODE REVIEW

File: app/src/main/java/org/fossasia/susi/ai/chat/adapters/recycleradapters/VoiceCommandsAdapter.kt [hunk 1]
```diff
@@ -22,6 +24,17 @@ class VoiceCommandsAdapter(val items: ArrayList<String>, val context: Context?)
...
+            val chatMessage = items.get(position)
+            val splits = chatMessage.split("\n".toRegex()).dropLastWhile { it.isEmpty() }
+            val message = splits.joinToString(" ")
+            if (!chatMessage.isEmpty()) {
+                chatPresenter.sendMessage(message, chatMessage)
+                val intent = Intent(context, ChatActivity::class.java)
+                intent.flags = Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_NEW_TASK
+                context?.startActivity(intent)
+            }
+        }
```
Comment (arundhati24) [inline app/src/main/java/org/fossasia/susi/ai/chat/adapters/recycleradapters/VoiceCommandsAdapter.kt]: A better place for this code is inside the ViewHolder class. Inside the view holder class, the data is bound with the view while the view is bound with the holder in the onBindViewHolder().

Comment (atm1504) [reply app/src/main/java/org/fossasia/susi/ai/chat/adapters/recycleradapters/VoiceCommandsAdapter.kt]: Initially, it was inside ViewHolder class, but I found that it was not working as expected. So, I moved the code inside the BIndViewHolder. I debugged it and found, the onClick was simply not recognizable.

Comment (arundhati24) [reply app/src/main/java/org/fossasia/susi/ai/chat/adapters/recycleradapters/VoiceCommandsAdapter.kt]: You should try debugging and looking into why the code was not working.

Comment (atm1504) [reply app/src/main/java/org/fossasia/susi/ai/chat/adapters/recycleradapters/VoiceCommandsAdapter.kt]: I tried doing so. Also, I rewrote the code differently, but the result was the same. So, upon google search, I came upon this kind of solution.

Comment (arundhati24) [reply app/src/main/java/org/fossasia/susi/ai/chat/adapters/recycleradapters/VoiceCommandsAdapter.kt]: Google search will definitely guide you, but at the end of the day, you will have to debug it line by line yourself and figure out what's going wrong. I suggest you check the code once again. Check the errors carefully, read about those errors i.e. why and when are they encountered and then find an appropriate solution for the same and fix it. That's how you should approach.

Comment (atm1504) [reply app/src/main/java/org/fossasia/susi/ai/chat/adapters/recycleradapters/VoiceCommandsAdapter.kt]: I will surely try debugging it again and provide you the updates.

**Comment by atm1504** [comment]
@arundhati24 @iamareebjamal @batbrain7 please review it.
@arundhati24 I have debugged and fixed the error.

**Comment by atm1504** [comment]
@arundhati24 @iamareebjamal as the pr is passing the grade tests locally, but failing online, it seems there is some problem in the circle ci. So, I am closing the pr and opening it from a new branch to trigger it.

**Pull request closed without merging**